presto-lark-sheets/src/main/java/com/facebook/presto/lark/sheets/LarkSheetsMetadata.java

```diff
@@ -0,0 +1,343 @@
+            }
+        }
+        return builder.build();
+    }
+
+    @Override
+    public void createSchema(ConnectorSession session, String schemaName, Map<String, Object> properties)
+    {
+        String token = LarkSheetsSchemaProperties.getSchemaToken(properties).orElseThrow(
+                () -> new PrestoException(SCHEMA_TOKEN_NOT_PROVIDED, "Schema token is required but not provided"));
+        boolean publicVisible = LarkSheetsSchemaProperties.isSchemaPublic(properties);
+        schemaStore.insert(new LarkSheetsSchema(schemaName, session.getUser(), token, publicVisible));
+    }
+
+    @Override
+    public void dropSchema(ConnectorSession session, String schemaName)
+    {
+        LarkSheetsSchema schema = requireVisibleSchema(session, schemaName);
+        checkSchemaUpdatable(schema, session.getUser(), "drop");
+        schemaStore.delete(schemaName);
+    }
+
+    @Override
+    public void renameSchema(ConnectorSession session, String source, String target)
+    {
+        LarkSheetsSchema schema = requireVisibleSchema(session, source);
+        checkSchemaUpdatable(schema, session.getUser(), "rename");
+        schemaStore.rename(source, target);
+    }
+
+    private Optional<LarkSheetsSchema> getVisibleSchema(ConnectorSession session, String schemaName)
+    {
+        return schemaStore.get(schemaName)
+                .filter(schema -> schema.isVisibleTo(session.getUser()));
+    }
+
+    private LarkSheetsSchema requireVisibleSchema(ConnectorSession session, String schemaName)
+    {
+        return getVisibleSchema(session, schemaName)
+                .orElseThrow(() -> new PrestoException(SCHEMA_NOT_EXISTS,
+                        format("Schema %s not exists or not visible", schemaName)));
+    }
+
+    private void checkSchemaUpdatable(LarkSheetsSchema schema, String operationUser, String operation)
+    {
+        if (!schema.getUser().equalsIgnoreCase(operationUser)) {
+            throw new PrestoException(NOT_PERMITTED,
+                    format("User '%s' is not permitted to perform '%s' on schema '%s'", operationUser, operation, schema.getName()));
+        }
+    }
+
+    private void checkSchemaReadable(LarkSheetsSchema schema)
+    {
+        if (!api.isReadable(schema.getToken())) {
+            throw new PrestoException(SCHEMA_NOT_READABLE,
+                    format("Spreadsheet %s not readable", schema.getToken()));
+        }
+    }
+
+    private List<LarkSheetsColumnHandle> getColumns(LarkSheetsTableHandle table)
+    {
+        List<String> header = api.getHeaderRow(table.getSpreadsheetToken(), table.getSheetId(), table.getColumnCount());
+
+        int numColumns = header.size();
+        LinkedHashMap<String, LarkSheetsColumnHandle> columns = new LinkedHashMap<>(numColumns);
+        for (int i = 0; i < numColumns; i++) {
+            String rawColumnName = header.get(i);
+            if (rawColumnName == null) {
```

Comment:
I'm just thinking if we could use the column index when the column name is null.   Your call.

Reply:
The column name can be a sequence of any UTF-8 characters. So it might be hard to define the column name format. For example, using `$2` as the column index to reference the second column, it might conflict with another column with the name of `$2`. In the other hand, updating a spreadsheet to fill a column name is very easy for end users. So column index name is not be supported.

Reply:
gotcha, thanks for the clarification!